flytekit/core/type_engine.py

```diff
@@ -2194,6 +2304,34 @@ def _check_and_covert_float(lv: Literal) -> float:
     raise TypeTransformerFailedError(f"Cannot convert literal {lv} to float")
 
 
+def _handle_flyte_console_float_input_to_int(lv: Literal) -> int:
+    """
+    Flyte Console is written by JavaScript and JavaScript has only one number type which is float.
```

Comment:
technically javascript's number type is [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) but yeah, sometimes it keeps track of trailing 0s and sometimes it doesn't.

Reply:
thank you, it's important to make words more accurately